src/pages/Pool/AddLiquidity.js

```diff
@@ -61,6 +68,27 @@ class AddLiquidity extends Component {
     return `Balance: ${value.dividedBy(10 ** decimals).toFixed(4)}`;
   }
 
+  isUnapproved() {
+    const { account, exchangeAddresses, selectors } = this.props;
+    const { outputCurrency } = this.state;
+
+    if (!outputCurrency) {
+      return false;
+    }
+
+    const { value, label } = selectors().getApprovals(
+      outputCurrency,
+      account,
+      exchangeAddresses.fromToken[outputCurrency]
+    );
+
+    if (!label || value.isLessThan(BN(10 ** 22))) {
```

Comment:
Everytime a transformFrom happened, the approvals amount will decrease.

Which means that this will appear as `unapproved` whenever user made one swap, regardless of how little, because it will change `allowance` to below `10 ** 22`

We should compare this with `inputValue` and `outputValue` instead.